### ISSUE TITLE
imgui.input_text boxes deselect, preventing text input

### ISSUE THREAD

**Issue by LittleSisterLover** [praydog/REFramework#1050]
**Describe the bug**
imgui.input_text deselects seemingly on frame, preventing input completely.

**Upload logs and any crash dumps**
[re2_framework_log.txt](https://github.com/praydog/REFramework/files/15137030/re2_framework_log.txt)

**Identify the REFramework version**
First noticed on Nightly842, still present in Nightly869 and Nightly878. Is not present on Nightly802-bugfix.

**To Reproduce**
Run an REFramework Lua script using the imgui.input_text function within its generated UI, attempt to click the box and observe that it does not stay selected.

**Expected behavior**
Text box should remain selected until clicked outside of it.

**Desktop and VR (please complete the following information):**
 - OS (e.g. Windows 11) - Windows 10
 - Graphics card - RTX 4070 Ti
 - CPU - i9-13900k

**Comment by praydog** [comment]
Cannot reproduce. Please provide a sample script. Try removing your other scripts.

**Comment by LittleSisterLover** [comment]
Here's a simple test script in which the issue occurs:
[imguitestt.zip](https://github.com/praydog/REFramework/files/15138634/imguitestt.zip)

Further testing has shown that this only occurs in Nightly878 if the value passed through the "label" parameter (as defined here [https://cursey.github.io/reframework-book/api/imgui.html](url)) is an empty string. 

This was not the case in Nightly802-bugfix, wherein any string passed through the "label" parameter would work.

**Comment by praydog** [comment]
Use imgui.push/pop_id around empty input boxes to fix this. This doesn't seem like a bug, and was likely fixed when ImGui was upgraded to 1.90.1. If anything, it seemed like a bug before that this *wasn't* happening.

https://github.com/ocornut/imgui/blob/master/docs/FAQ.md#q-about-the-id-stack-system

Example:
```lua
imgui.push_id("My unique Identifier")
local changed, value = imgui.input_text("", display_text)
-- logic
imgui.pop_id()
```

**Comment by LittleSisterLover** [comment]
Interesting. I was using push/pop_id for identifiers during loops, but didn't consider it here. If I'm following correctly, the label is used as the identifier for text_input. An empty string leaves it without an one, which causes the behaviour. Pushing/popping an ID then puts one in its place, allowing it to work. There must've been a change in the update to how identifiers are handled that leads to an empty string no longer being a valid identifier.

Regardless, thank you for both looking into this and providing a reference.
